### PR TITLE
Fixes for bugs encountered in during test gyms playthroughs

### PR DESCRIPTION
**Pull request by m-samiec**
#### Description
Bug fixes I encountered when doing the replication graph work. Pulling into a separate PR so they can get merged now.

Fixes dormant actor deletion when using NCD interest
Fixes an issue where load balancer would crash when using world composition on clients
Fixes an issue with the load balancer printing bad warnings when using world composition

#### Release note
Transient fixes, not release note

#### Tests
Ran through test gyms

#### Primary reviewers
@jacquese @mironec @alastairdglennie

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp [hunk 1]
```diff
@@ -456,6 +456,12 @@ TArray<FWorkerComponentData> EntityFactory::CreateTombstoneEntityComponents(AAct
 	Components.Add(Tombstone().CreateData());
 	Components.Add(EntityAcl(ReadAcl, WriteAclMap()).CreateEntityAclData());
 
+	Worker_ComponentId ActorInterestComponentId = ClassInfoManager->ComputeActorInterestComponentId(Actor);
```
Comment (jacquese) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp]: What is this doing? Is this a marker component that identifies this actor?

**Comment by mironec** [comment]
Can we also include the fix for the crash when updating components to unmapped? Will get the code momentarily.

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 2]
```diff
@@ -657,13 +657,22 @@ void USpatialNetDriver::MakePlayerSpawnRequest()
...
+		*GetNameSafe(LoadedLevel), *GetNameSafe(OwningWorld), *GetNameSafe(World));
+
 	// Callback got called on a World that's not associated with this NetDriver.
 	// Don't do anything.
 	if (OwningWorld != World)
 	{
 		return;
 	}
 
+	// Not necessary for clients
```
Comment (jacquese) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: 🎉

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 3]
```diff
@@ -657,13 +657,22 @@ void USpatialNetDriver::MakePlayerSpawnRequest()
...
+	if (!IsServer())
+	{
+		return;
+	}
+
```
Comment (jacquese) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: 🎉

File: SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp (hunk 1 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp]: Yep, same logic is run above when create a normal entity

**Pull request merged**